### PR TITLE
rework chat user list, compact, add swords

### LINKED ISSUE #1221: Rework chat player list
Most users want a more compact, faster/more useable user list. Swap the Playing/waiting text with a sword icon like in the old client on the right (preserves space, easier to spot than reading text).

The player list displays the currently joined/hosted/playing game.
The old client displayed a map preview there aswell.

- [ ] compact (remove second line)
- [ ] add playing/waiting (sword) status symbol
- [ ] display game map icon

http://forums.faforever.com/viewtopic.php?f=2&t=17068&start=70#p173784
http://forums.faforever.com/viewtopic.php?f=2&t=17068&start=40#p172187

### PR DESCRIPTION
**Pull request by Geosearchef**
the majority of the user base favors a compact player list, the "playing"/"waiting"/"hosting" text is replaced by the sword symbols from the python client to reduce space consumption and improve legibility.

closes #1221

image

### CODE REVIEW

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChannelTabController.java]: EHMM GEO????

Comment (1-alex98) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: What does this fix I am certain there is a clean way for this

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChatUserItemController.java]: no commented out code

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChatUserItemController.java]: u know no commented out code

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChatUserItemController.java]: I don't think we wanna load an image per player. Can we not only load the three

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChatUserItemController.java]: If we do it at initialize will it update?

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Comment by 1-alex98** [comment]
Also misses unit test and travis fails

**Comment by 1-alex98** [comment]
Should be easy to unit test the init method

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: It fixes the user list to a fixed width instead of specifying that as a percentage of the monitor size. And yes, I have searched thoroughly, there is no other way to do this.

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChatUserItemController.java]: ohhhhhhhhhh, great that you caught that, I didn't realize this is part of the user item controller, ooops

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChatUserItemController.java]: what do you mean?

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (1-alex98) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: ok fine

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/ChannelTabController.java]: UI values should not be hardcoded. Think about theming.
Also we do not address sizes in pixels, but in relative width.

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/ChannelTabController.java]: If required, use log.debug and a proper message text.

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/ChannelTabController.java]: If you want to access the width I suggest you read it from the UI control (e.g. min width or wherever it makes sense to set it).

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/ChatUserItemController.java:332]: Do the i18n string need to be removed from the translation files?

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChannelTabController.java]: No system out println. log correctly

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChatUserItemController.java]: commeted out code

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChatUserItemController.java]: Where is that set?

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChatUserItemController.java]: yes, that's intentional, this is so far not supported by the client

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/ChatUserItemController.java]: Instead of new ClassPathresource this should be `uiService.getThemeFile` each

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/ChatUserItemController.java:490]: The whole block could use Optional's ifPresentOrElse

File: src/main/resources/theme/chat/channel_tab.fxml
Comment (Brutus5000) [inline src/main/resources/theme/chat/channel_tab.fxml]: I don't think comments in FXML are a good idea anyway. Might get lost due to SceneBuilder logic anytime, also you don't see it in SceneBuilder. I talked about the fixed px above already.

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (1-alex98) [reply src/main/java/com/faforever/client/chat/ChatUserItemController.java]: Then make a comment explaining that

**Comment by Geosearchef** [comment]
unit test build fixed, unit tests switched to new status impl

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: It makes no sense to set the width of the user list as a percentage as it's wasting chat space and seperating users from the status indicator.

File: src/test/java/com/faforever/client/chat/ChatChannelUserItemControllerTest.java
Comment (1-alex98) [inline src/test/java/com/faforever/client/chat/ChatChannelUserItemControllerTest.java:215]: assertThat(...,is(false))

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/test/java/com/faforever/client/chat/ChatChannelUserItemControllerTest.java
Comment (Geosearchef) [reply src/test/java/com/faforever/client/chat/ChatChannelUserItemControllerTest.java:215]: why the heck should I do that? There's a method for that purpose. Even intellij automatically detects that and advises me to convert to assertFalse.

Comment (1-alex98) [reply src/test/java/com/faforever/client/chat/ChatChannelUserItemControllerTest.java:215]: well ok then

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (1-alex98) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: Well you can as he said below declare it in the fxml at some other place

Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: How should I declare that value in fxml?

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: bad idea, that screws with the UI as this value is changing as the UI changes.

File: src/main/java/com/faforever/client/chat/ChatUserItemController.java
Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChatUserItemController.java:490]: making the entire block less readable, yes

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (1-alex98) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: @Brutus5000

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (Brutus5000) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: Well the new implementation also screws it on resizing.

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (Brutus5000) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: I though about using pref width or min width on the control. But it looks like it is loosing these values during runtime. Imho we need a solution that does not require this code-wise behavior.

**Comment by 1-alex98** [comment]
Still travis fails

**Comment by Geosearchef** [comment]
Did you take a look why? Afaik not due to the PR itself.

**Comment by 1-alex98** [comment]
It is no unit test failing. It does not build we can not merge this unless it builds

File: src/main/java/com/faforever/client/chat/ChannelTabController.java
Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: due to it's somewhat not working nature I've completely reset this code

Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChannelTabController.java]: Not used any longer

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Comment by 1-alex98** [comment]
@Geosearchef will you rework or shall i?

**Comment by 1-alex98** [comment]
@Brutus5000 ok now?

**Comment by Brutus5000** [comment]
Address codacy issues, please @axel1200

**Review by 1-alex98 — APPROVED** [review]

**Pull request merged**
